### DISCUSSION TITLE
Key Extraction

### DISCUSSION THREAD

**Discussion by nachtfunke** [vinissimus/next-translate#790] [Q&A] [answered]
Hello! I am not sure if the project supports this or not, but manually maintaining keys quickly becomes error prone. What is this plugin's preferred way to automatically extract set keys into the locale jsons?

**Comment by aralroca** [comment]
You can use the `logger` function in the configuration to catch these missing keys: https://github.com/vinissimus/next-translate#3-configuration

**Reply by nachtfunke** [reply]
This is helpful, but it doesn't really help me, automatically extracting keys, it just tells me about missing keys. The issue here is though, that in order to maintain readability and the textual semantic, the keys sometimes are identical to their value. I am really looking for a way, to extract the keys.

**Reply by aralroca** [reply]
Maybe I misunderstood you. What do you mean by extracting the keys?

**Reply by nachtfunke** [reply]
I'm new to this library. I am used to working with other libraries that offer a way to parse files and generate locales from that step: https://lingui.js.org/ref/cli.html#extract-template for example - but the project I am working on just recently switched to this one.

At the moment, whenever a new key is used in the react code, I have to _manually_ add them to the `.json` of the language. In the project I work on I have already seen that not only are locales incongruent in their order but they also do not hold the same number of keys. A problem resulting of having to manually maintain they keys instead of generating them automatically from the templates for each locale.

Also, maybe the amount of text is just not maintainable with this approach of internationalisation - I am just researching the possibilities.

**Comment by wuifdesign** [comment, marked as answer]
i created a package for this purpose, hope it works fine.
https://www.npmjs.com/package/next-translate-scanner

**Reply by nachtfunke** [reply]
That's fantastic! I'll give this a try :)
